website/docs/secrets/managing-secrets-with-sops.mdx

```diff
@@ -0,0 +1,227 @@
+
+1- Generate a gpg key pairs
+
+```bash
+export KEY_NAME="gpg-key"
+export KEY_COMMENT="gpg key"
+
+gpg --batch --full-generate-key <<EOF
+%no-protection
+Key-Type: 1
+Key-Length: 4096
+Subkey-Type: 1
+Subkey-Length: 4096
+Expire-Date: 0
+Name-Comment: ${KEY_COMMENT}
+Name-Real: ${KEY_NAME}
+EOF
+```
+
+2- Export the key pairs fingerprint in the shell
+```bash
+gpg --list-secret-keys "${KEY_NAME}"
+
+sec   rsa4096 2020-09-06 [SC]
+      710DC0DB6C1662F707095FC30233CB21E656A3CB
+      
+export KEY_FP="710DC0DB6C1662F707095FC30233CB21E656A3CB"
+```
+
+3- Export the generated private key to a kubernetes secret `sops-gpg-private-key` which will be used by flux's kustomize-controller to decrypt the secrets using sops.
+
+```bash
+gpg --export-secret-keys --armor "${KEY_FP}" | kubectl create secret generic sops-gpg-private-key --namespace=flux-system --from-file=sops.asc=/dev/stdin
+```
+
+4- Export the generated public key to a kubernetes secret `sops-gpg-public-key` which will be used by Weave GitOps Enterprise to encrypt the secrets created from the UI.
+
+```bash
+gpg --export --armor "${KEY_FP}" | kubectl create secret generic sops-gpg-public-key --namespace=flux-system --from-file=sops.asc=/dev/stdin
+```
+
+> It's recommended to remove the secret from your machine
+
+```bash
+gpg --delete-secret-keys "${KEY_FP}"
+```
+
+5- Create a kustomization for reconciling the secrets on the cluster and set the `--decryption-secret` flag to the name of the private key created in step 3.
+
+```bash
+flux create kustomization gpg-secrets \
+--source=secrets \
+--path=./secrets/gpg \
+--prune=true \
+--interval=10m \
+--decryption-provider=sops \
+--decryption-secret=sops-gpg-private-key
+```
+
+6- Annotate the kustomization object created in the previous step with the name and namespace of the public key created in step 4.
+
+```bash
+kubectl annotate kustomization gpg-secrets sops-public-key/name=sops-gpg-public-key sops-public-key/namespace=flux-system
+```
+
+The kustomization object should looks like this
+
+```yaml
+apiVersion: kustomize.toolkit.fluxcd.io/v1beta2
+kind: Kustomization
+metadata:
+  name: gpg-secrets
+  namespace: flux-system
+  annotations:
+    sops-public-key/name: sops-gpg-public-key
+    sops-public-key/namespace: flux-system
+spec:
+  interval: 10m
+  sourceRef:
+    kind: GitRepository
+    name: secrets
+  path: ./secrets/gpg
+  decryption:
+    provider: sops
+    secretRef:
+      name: sops-gpg-private-key
+  prune: true
+  validation: server
+```
+
+> This is an essential step in order to allow other operators and developers to utilize WeaveGitOps UI to encrypt SOPS secrets using the public key secret in the cluster.
+
+## Encrypting secrets using age
+
+[age](https://github.com/FiloSottile/age) is a simple, modern and secure file encryption tool, It can be used to encrypt secrets using Weave GitOps Enterprise.
```

Comment:
This should either be `age is a simple, modern and secure file encryption tool. It can be used to encrypt secrets using Weave GitOps Enterprise.` or
`age is a simple, modern and secure file encryption tool, that can be used to encrypt secrets using Weave GitOps Enterprise.`
But the current punctuation is not correct.